### PR TITLE
refactor(api):  engine core move_to additional arguments

### PR DESCRIPTION
**Pull request by jbleon95**
# Overview

Addresses RCORE-324.

This PR adds the additional optional arguments `force_direct`, `minimum_z_height` and `speed` to the Protocol Engine core implementation of `move_to`. These arguments have been added to the requisite protocol engine commands, as well as the sync client and other related methods.

# Changelog
- `force_direct`, `minimum_z_height` and `speed` implemented in PAPIv2 engine core
- `forceDirect`, `minimumZHeight` and `speed` added to Protocol Engine `move_to_well`
- `speed` added to Protocol Engine `move_to_coordinates`
- Refactored `get_movement_waypoints_to_well`

# Review requests
PAPIv2 engine core implementation of `move_to` correctly accepts new arguments, and Protocol Engine `move_to_well` and `move_to_coordinates` work with new parameters.

# Risk assessment
Medium, refactor of waypoint motion planning logic and additional arguments added to `move_to` methods, but arguments are optional and defaults do not change existing behavior, and all new changes are well covered by unit tests.

### CODE REVIEW

File: api/src/opentrons/protocol_engine/execution/movement.py [hunk 1]
```diff
@@ -128,9 +131,12 @@ async def move_to_well(
             origin_cp=origin_cp,
             max_travel_z=max_travel_z,
             current_well=current_well,
+            force_direct=force_direct,
+            minimum_z_height=minimum_z_height,
         )
 
-        speed = self._state_store.pipettes.get_movement_speed(pipette_id=pipette_id)
+        if not speed:
+            speed = self._state_store.pipettes.get_movement_speed(pipette_id=pipette_id)
```
Comment (jbleon95) [inline api/src/opentrons/protocol_engine/execution/movement.py]: look into having requested speed be an argument of get_movement speed, remove `if` and future proof handling user given speed

File: api/src/opentrons/protocol_engine/state/motion_utils.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+"""Getters for Protocol Engine motion planning."""
```
Comment (jbleon95) [inline api/src/opentrons/protocol_engine/state/motion_utils.py]: rename file `move_type.py`

File: api/tests/opentrons/protocol_engine/state/test_motion_view.py [hunk 3]
```diff
@@ -39,6 +35,13 @@ def patch_mock_get_waypoints(decoy: Decoy, monkeypatch: pytest.MonkeyPatch) -> N
     monkeypatch.setattr(motion_planning, "get_waypoints", mock_get_waypoints)
 
 
+@pytest.fixture(autouse=True)
+def patch_mock_get_move_type(decoy: Decoy, monkeypatch: pytest.MonkeyPatch) -> None:
+    """Replace motion_utils.get_waypoints() with a mock."""
+    mock_get_move_type = decoy.mock(func=motion_utils.get_move_type_to_well)
```
Comment (jbleon95) [inline api/tests/opentrons/protocol_engine/state/test_motion_view.py]: search for `inspect.get_members` and use that instead

**Review by mcous — APPROVED** [review]
🎥

**Pull request merged**